src/go/rpk/pkg/config/weak_test.go

```diff
@@ -0,0 +1,429 @@
+package config
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+func TestWeakBool(t *testing.T) {
+	type testWeakBool struct {
+		Wb weakBool `yaml:"wb"`
+	}
+
+	tests := []struct {
+		name    string
+		data    string
+		expBool bool
+		expErr  bool
+	}{
+		{
+			name:    "should unmarshal boolean:true",
+			data:    "wb: true",
+			expBool: true,
+			expErr:  false,
```

Comment:
I'd leave out the expErr field when we aren't expecting one, and rely on the default false. If we expect a value, we shouldn't specify `expErr: false` -- it's easier to read when I _am_ expecting an error in these tests.